core/chains/evm/logpoller/log_poller.go

```diff
@@ -1344,6 +1345,6 @@ func EvmWord(i uint64) common.Hash {
 	return common.BytesToHash(b)
 }
 
-func (lp *logPoller) FilteredLogs(queryFilter query.KeyFilter, sortAndLimit query.LimitAndSort) ([]Log, error) {
-	return lp.orm.FilteredLogs(queryFilter, sortAndLimit)
+func (lp *logPoller) FilteredLogs(ctx context.Context, queryFilter query.KeyFilter, sortAndLimit query.LimitAndSort, mapper *EventFilterMapper) ([]Log, error) {
```

Comment:
Why do we need `EventFilterMapper`, all of this information should be in the filters. I wrote a function that maps chain agnostic filters to chain specific ones

Reply:
I did this because I needed to consolidate the query remapping with the ORM visitor in the same package to avoid circular dependencies.

Reply:
Thats too bad, I think the remapping should stay out of log poller, must be some other approach to avoid circular dependency here?